packages/cosmic-swingset/provisioning-server/src/ag_pserver/main.py

```diff
@@ -88,32 +88,32 @@ class ConfigElement(Element):
     @staticmethod
     def gatherArgs(opts):
         meta = {}
-        f = open(cosmosConfigFile(opts['home']))
-        config = f.read()
+        with open(cosmosConfigFile(opts['home'])) as f:
+            config = f.read()
         gr = '/usr/src/app/lib/git-revision.txt'
         if os.path.exists(gr):
-            f = open(gr)
-            meta['package_git'] = f.read().strip()
+            with open(gr) as f:
+                meta['package_git'] = f.read().strip()
         else:
-            f = os.popen('git rev-parse --short HEAD')
-            sha = f.read().strip()
-            f = os.popen('git diff --quiet || echo -dirty')
-            meta['package_git'] = sha + f.read().strip()
+            with os.popen('git rev-parse --short HEAD') as f:
```

Comment:
huh, it hadn't occurred to me that `os.popen` provides a context manager, but I guess it makes sense